routers/routes/routes.go

```diff
@@ -513,8 +513,9 @@ func RegisterRoutes(m *macaron.Macaron) {
 		})
 	}, ignSignIn)
 
-	m.Group("", func() {
-		m.Post("/attachments", repo.UploadAttachment)
+	m.Group("/attachments", func() {
+		m.Post("", repo.UploadAttachment)
```

Comment:
Why we get repo attachments for update and delete, but issue/comment attachments for get? I understand that there's a special case for issues/comments under creation, but attachments can also be modified on edit.

By the way, this is a great opportunity to validate user permissions when updating/deleting an attachment associated to an existing issue/comment that they don't have access to. The criteria should be the same as for edit/delete the issue/comment.

Perhaps this is a matter for another PR.

Reply:
Yes, right checking for deleting attachment should definitely be done

Reply:
@guillep2k 
I use `[issue/comment]/attachments` because I could get attachments information without inject information into `view_content.tmpl`. Otherwise I have to inject attachments name, uuid and size during rendering `view_content.tmpl` for initializing **dropzone** when first time **edit button** click. 

However, as you said it's okay to modify attachments on edit. Do you think I should modify my codes? If there is anything I misunderstand, please let me know 😃 

@lafriks 
I think I could try to enable user permissions checking, but I think it's better to open another pull request.

Reply:
> 
> 
> @guillep2k
> I use `[issue/comment]/attachments` because I could get attachments information without inject information into `view_content.tmpl`. Otherwise I have to inject attachments name, uuid and size during rendering `view_content.tmpl` for initializing **dropzone** when first time **edit button** click.
> 
> However, as you said it's okay to modify attachments on edit. Do you think I should modify my codes? If there is anything I misunderstand, please let me know 😃
> 
> @lafriks
> I think I could try to enable user permissions checking, but I think it's better to open another pull request.

I think that changing the API is not required per se, but it would be needed when access permissions were to be checked, so it can wait for the next PR.

Reply:
@guillep2k 
Ok, I'll finish refactoring `UpdateCommentContent` and `UpdateIssueContent`. Then moving on checking access permissions. 👍

Reply:
Problem with delete is that it introduces vulnerability that knowing attachment uuid you could delete ones you do not have rights for

Reply:
@lafriks it's worse: you can update it!

Reply:
@lafriks 
You are right, I'll check user right for deleting attachments on next push in advance.

Reply:
We should at least maybe add the uploader userid if known and allow deletion of it is the same or an admin ?
Some work for security is done on the subject on #7956 but I remember that some test still doesn't pass and haven't much time to work on it.

Reply:
Hummm... I see the problem. I underestimate the effort to check attachment permission. No wonder Gitea does not actually delete file on server when removing it from **dropzone**.
 
So I guess the possible solution may be either:

1. Not to delete attachment on server actually during editing for short term. Since `UpdateIssueContent` and `UpdateCommentContent` will check login user, there is less security concern IMO.
2. Keep working on #7956 to check attachment's uploader before deleting.

Correct me if I'm wrong. 😕

Reply:
But isn't the same problem present in update as well? I think that only repo access is checked when you know the attachment GUID.

Reply:
Let me recap:
1. My `/attachments/delete` will cause security issue because I only check whether user is login with:
  ```go
  m.Group("/attachments", func() {
    m.Post("", repo.UploadAttachment)
    m.Post("/delete", repo.DeleteAttachment)
  }, reqSignIn)
  ```
  The way to fix it is either:
  - Remove `/attachments/delete` route, attachments will not be deleted on server just as usual when creating new issue/comment
  - At least check Uploader ID in `repo.DeleteAttachment` such as
    ```go
    func DeleteAttachment(ctx *context.Context) {
      file := ctx.Query("file")
      attach, err := models.GetAttachmentByUUID(file)
      if attach.UploaderID != ctx.User.ID {
        ctx.Error(403)
      }
      ...
    }
    ```
    But I see there is a comment in:
      ```go
      type Attachment struct {
        ID            int64  `xorm:"pk autoincr"`
        UUID          string `xorm:"uuid UNIQUE"`
        IssueID       int64  `xorm:"INDEX"`
        ReleaseID     int64  `xorm:"INDEX"`
        UploaderID    int64  `xorm:"INDEX DEFAULT 0"` // Notice: will be zero before this column added
        CommentID     int64
        Name          string
        DownloadCount int64              `xorm:"DEFAULT 0"`
        Size          int64              `xorm:"DEFAULT 0"`
        CreatedUnix   timeutil.TimeStamp `xorm:"created"`
      }
      ```
    So maybe UploaderID is not available yet?
2. `Issue.UpdateAttachments` may also casue security issue(?)
  It seems to me that I use it only inside `repo.UpdateIssueContent` and
  user permission will be checked in the begin of `repo.UpdateIssueContent`. So
  maybe it's okay?
3. Other attachments' security issue requires finishing #7956 to solve

Correct me if I'm wrong. Thank you~

Reply:
You are correct 😉 
The same problems delete introduce are already present with update. What can happen is the following:

User **bob** creates an issue with attachments `abcd` and `efgh`. **Alice** is a little mad with **bob** (reasons unknown), so she decides to play him a joke. She's skilled in Javascript, so she opens the issue, starts a new comment and POST content into `abcd`. Now the attachment is modified. 😢. The same can happen with the new delete option, as it copies the previous behavior.

Back to your options: I think we should decide whether solving this is a matter of this PR or another. Considering that update was already affected, perhaps we can treat this as a different problem. I forgot about @sapk PR. It's reasonable to expect this to be resolved there.

@lafriks  What do you think? @sapk?

Reply:
> After trying, it seems like the UploaderID is not correctly set yet (using sqlite)

I found I'm wrong, UploaderID is correctly set (using sqlite). It turns out it's my codes in `index.js` is wrong.
```js
$.post($dropzone.data('remove-url'), {
    file: file.uuid,
    _csrf: $dropzone.data('csrf')
});
```
I used `file.uuid` for **POST** request, but it will not be set. `filenameDict[file.name].uuid` should be used instead.

So I think checking `UploaderID` before delete attachment could be a short-term workaround.
I'll update my codes later. 😓